src/Azure.Functions.Cli/Kubernetes/KubernetesHelpers.cs

```diff
@@ -146,18 +165,26 @@ internal static string GetKedaResources(string @namespace)
             int? cooldownPeriod = null,
             string serviceType = "LoadBalancer",
             int? minReplicas = null,
-            int? maxReplicas = null)
+            int? maxReplicas = null,
+            string keysSecretCollectionName = null,
+            bool mountKeysAsContainerVolume = false)
         {
             ScaledObjectV1Alpha1 scaledobject = null;
             var result = new List<IKubernetesResource>();
             var deployments = new List<DeploymentV1Apps>();
             var httpFunctions = triggers.FunctionsJson
                 .Where(b => b.Value["bindings"]?.Any(e => e?["type"].ToString().IndexOf("httpTrigger", StringComparison.OrdinalIgnoreCase) != -1) == true);
             var nonHttpFunctions = triggers.FunctionsJson.Where(f => httpFunctions.All(h => h.Key != f.Key));
+            keysSecretCollectionName = string.IsNullOrEmpty(keysSecretCollectionName)
+                ? $"func-keys-kube-secret-{name}"
+                : keysSecretCollectionName;
             if (httpFunctions.Any())
             {
                 int position = 0;
                 var enabledFunctions = httpFunctions.ToDictionary(k => $"AzureFunctionsJobHost__functions__{position++}", v => v.Key);
+                //Environment variables for the func app keys kubernetes secret
+                var kubernetesSecretEnvironmentVariable = FuncAppKeysHelper.FuncKeysKubernetesEnvironVariables(keysSecretCollectionName, mountKeysAsContainerVolume);
+                enabledFunctions = enabledFunctions.Concat(kubernetesSecretEnvironmentVariable).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
```

Comment:
```suggestion
                var additionalEnvVars = enabledFunctions.Concat(kubernetesSecretEnvironmentVariable).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
```

and replace `enabledFunctions` in the `GetDeployment()` call